core/client/controllers/settings/navigation.js

```diff
@@ -1,23 +1,111 @@
+    })
+});
+
+NavigationController = Ember.Controller.extend({
+     navigationItems: Ember.computed('model.navigation', function () {
+        var navItems,
+            lastItem;
+
+        try {
+            navItems = JSON.parse(this.get('model.navigation') || [{}]);
+        } catch (e) {
+            navItems = [{}];
+        }
+
+        navItems = navItems.map(function (item) {
+            return NavItem.create(item);
+        });
+
+        lastItem = navItems.get('lastObject');
+        if (!lastItem || !lastItem.get('isBlank')) {
+            navItems.addObject(NavItem.create());
+        }
+
+        return navItems;
+    }),
+
+    navigationItemsObserver: Ember.observer('navigationItems.[]', function () {
+        var navItems = this.get('navigationItems');
+
+        navItems.forEach(function (item, index, items) {
+            if (index === (items.length - 1)) {
+                item.set('last', true);
+            } else {
+                item.set('last', false);
+            }
+        });
     }),
 
     actions: {
         addItem: function () {
-            // Add a new item
+            var navItems = this.get('navigationItems'),
+                lastItem = navItems.get('lastObject');
+
+            if (lastItem && !lastItem.get('isBlank')) {
+                navItems.addObject(NavItem.create());
+            }
         },
 
-        deleteItem: function () {
-            // Delete navItem which should be a function param like: `deleteItem: function(navItem) {`
+        deleteItem: function (item) {
+            if (!item) {
+                return;
+            }
+
+            this.get('navigationItems').removeObject(item);
         },
 
         save: function () {
-            // Save everything
+            var self = this,
+                navSetting,
+                blogUrl = this.get('config').blogUrl,
+                blogUrlRegex = new RegExp('^' + blogUrl + '(.*)', 'i'),
+                match;
+
+            navSetting = this.get('navigationItems').map(function (item) {
+                var label,
+                    url;
+
+                if (!item || item.get('isBlank')) {
+                    return;
+                }
+
+                label = item.get('label').trim();
+                url = item.get('url').trim();
+
+                match = url.match(blogUrlRegex);
+
+                if (match) {
+                    if (match[1] === '') {
+                        url = '/';
+                    } else {
+                        url = match[1];
+                    }
+                } else if (!validator.isURL(url) && url !== '' && url[0] !== '/') {
+                    url = '/' + url;
+                }
+
+                return {label: label, url: url};
+            }).compact();
+
+            this.set('model.navigation', JSON.stringify(navSetting));
+            this.get('model').notifyPropertyChange('navigation');
```

Comment:
:+1: I'm with you now. Seems like a good thing to leave a comment on, for people who bump into this down the road, to make sure they don't delete the line.